src/main/java/com/google/devtools/build/lib/cmdline/Label.java

```diff
@@ -203,6 +211,31 @@ public static Label parseWithPackageContext(String raw, PackageContext packageCo
     return createUnvalidated(PackageIdentifier.create(repoName, pkgFragment), parts.target());
   }
 
+  public static final class RepoMappingRecorder {
```

Comment:
Please add some basic unit tests of the repo mapping recording functionality in https://cs.opensource.google/bazel/bazel/+/master:src/test/java/com/google/devtools/build/lib/cmdline/LabelTest.java.

Reply:
done!